src/types/params/plain_secret.rs

```diff
@@ -110,6 +110,7 @@ impl<'a> PlainSecretParamsRef<'a> {
         hasher.finalize().to_vec()
     }
 
+    /// Uses sha1_checked
```

Comment:
this means these functions panic now on a collision, right?

Reply:
I'll take a stab at adjusting all relevant functions to return a "collision detected" error, instead of doing the hash digest mitigation thing.

(I think that's what you had in mind?)

Reply:
I've adjusted almost all uses of `sha1-checked` to return a new `Error::HashCollision` if a collision is detected.

One intentional deviation from that pattern is fingerprint calculation: I think it is unreasonable to expect consumers of an OpenPGP library to handle the possibility that fingerprint calculation might error out.

My current feeling is that it's a better tradeoff to return the "mitigated" hash digest as a fingerprint instead of returning errors.

Reply:
sounds reasonable